### DISCUSSION TITLE
Is there a way to map pitch from an audio to another one ?

### DISCUSSION THREAD

**Discussion by Mikerhinos** [coqui-ai/TTS#3024] [General]
The concept is : take a song, translate lyrics, generate voice with these lyrics, apply pitch from original song (isolated voice) to the TTS generated voice, put back music.
It would probably be adjusted manually in some way because for example french is often longer to say something compared to english, but there surely has to be some starting way.

**Comment by digdemyildiz** [comment]
Were you able to find any solution ? :)

**Reply by Mikerhinos** [reply]
Unfortunately not :(

**Reply by digdemyildiz** [reply]
The concept you mentioned is our senior design project and we are having hard time finding tools to accomplish. If there are any other resources, we would be grateful if you can share. Thank you for your response, have a great week :)

**Reply by YuCaIb** [reply]
@digdemyildiz  I do have a smilar research project as well. Please leave a comment, if there's any development.

**Reply by Mikerhinos** [reply]
When I paused this "project" I found a Spotify tool that can extract a MIDI pitch file from an audio (https://github.com/spotify/basic-pitch) and the idea was to get original voice pitch file and then apply it to the generated TTS audio.
I didn't found a tool that was able to do it, it seems that studio audio tools like Cubase or something can do it, but I do not have a license so I wasn't able to experiment it.
There is also the problem of phoneme length (which made me pause this idea), because getting the pitch is a thing, but for a song you also have to get the words pronounced in the same time than the original, so if you take for example "Take On Me" by AHA some words can be quite long and the TTS will be short and constant :/

It must be possible because a little french startup is working on a tool they call Lipitt that can apparently work on songs (some examples https://twitter.com/search?q=lipitt&src=typed_query&f=live ).

**Reply by digdemyildiz** [reply]
Thanks a lot. We will review the resources. Unfortunately, we do not have a prior experience in NLP or Audio Processing, that's why we are searching for already implemented models, but we were not able find one too :(
